dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ranger/RangerConnector.java

```diff
@@ -0,0 +1,284 @@
+            try {
+              return handle.rangerClient.findGroups(toParameters(page));
+            } catch (RangerException e) {
+              throw new RuntimeException("Failed to fetch Ranger groups", e);
+            }
+          },
+          handle.pageSize);
+    }
+
+    @Override
+    protected String toCallDescription() {
+      return "Ranger groups";
+    }
+  }
+
+  static class DumpRolesTask extends AbstractRangerTask<Role> {
+
+    DumpRolesTask() {
+      super(RolesFormat.ZIP_ENTRY_NAME);
+    }
+
+    @Override
+    protected Iterator<Role> dataIterator(
+        @Nonnull Writer writer, @Nonnull RangerClientHandle handle) {
+      return new RangerPageIterator<>(
+          page -> {
+            try {
+              return handle.rangerClient.findRoles(toParameters(page));
+            } catch (RangerException e) {
+              throw new RuntimeException("Failed to fetch Ranger roles", e);
+            }
+          },
+          handle.pageSize);
+    }
+
+    @Override
+    protected String toCallDescription() {
+      return "Ranger roles";
+    }
+  }
+
+  static class DumpServicesTask extends AbstractRangerTask<Service> {
+
+    DumpServicesTask() {
+      super(ServicesFormat.ZIP_ENTRY_NAME);
+    }
+
+    protected Iterator<Service> dataIterator(
+        @Nonnull Writer writer, @Nonnull RangerClientHandle handle) {
+      return new RangerPageIterator<>(
+          page -> {
+            try {
+              return handle.rangerClient.findServices(toParameters(page));
+            } catch (RangerException e) {
+              throw new RuntimeException("Failed to fetch Ranger services", e);
+            }
+          },
+          handle.pageSize);
+    }
+
+    @Override
+    protected String toCallDescription() {
+      return "Ranger services";
+    }
+  }
+
+  static class DumpPoliciesTask extends AbstractRangerTask<Policy> {
+
+    DumpPoliciesTask() {
+      super(PoliciesFormat.ZIP_ENTRY_NAME);
+    }
+
+    @Override
+    protected Iterator<Policy> dataIterator(
+        @Nonnull Writer writer, @Nonnull RangerClientHandle handle) {
+      return new RangerPageIterator<>(
+          page -> {
+            try {
+              return handle.rangerClient.findPolicies(toParameters(page));
+            } catch (RangerException e) {
+              throw new RuntimeException("Failed to fetch Ranger policies", e);
+            }
+          },
+          handle.pageSize);
+    }
+
+    @Override
+    protected String toCallDescription() {
+      return "Ranger policies";
+    }
+  }
+
+  private abstract static class AbstractRangerTask<T> extends AbstractTask<Void> {
+
+    public AbstractRangerTask(String targetPath) {
+      super(targetPath);
+    }
+
+    @ForOverride
+    protected abstract Iterator<T> dataIterator(
+        @Nonnull Writer writer, @Nonnull RangerClientHandle handle) throws Exception;
+
+    @Override
+    protected Void doRun(TaskRunContext context, ByteSink sink, @Nonnull Handle handle)
```

Comment:
Consider adding nullability annotations to all arguments or only to `@Nullable` ones.